custom_logs.tf

```
@@ -0,0 +1,177 @@
+locals {
+  columns = {
+    "state"                  = "string"
+    "TimeGenerated"          = "datetime"
+    "additional_information" = "string"
+    "affected_entity"        = "string"
+    "affected_object"        = "string"
+    "monitor_description"    = "string"
+    "monitor_name"           = "string"
+    "monitor_package"        = "string"
+    "script_name"            = "string"
+    "script_version"         = "string"
+    "threshold"              = "string"
+    "value"                  = "string"
+  }
+  kql_types = join(", ", [for column_name, column_type in local.columns : "${column_name}:${column_type}"])
+
+  resource_group_id = substr(var.log_analytics_workspace.id, 0, length(regexall("^[^/]*/[^/]*/[^/]*/[^/]*/[^/]*", var.log_analytics_workspace.id)[0]))
```

```suggestion
  resource_group_id = regex("/resourceGroups/([^/]+)", var.log_analytics_workspace.id)[0]
```
Regex can do all of that in one function call. See https://developer.hashicorp.com/terraform/language/functions/regex

I haven't tested this regex pattern yet, but I think it should extract everything between `/resourceGroups/` and the next slash

---

the goal is getting everything before the resource group name as well, since it's full id is needed, i have not found a way to shorten the regex without including the slash after the resource group's name.

---

Ahh. I forgot about that. In that case how about:
```suggestion
  resource_group_id = regex("^((/[^/]+){4})", var.log_analytics_workspace.id)[0]
```
This basically what you are doing just in a loop instead of writing it out.